src/node.cc

```diff
@@ -2553,18 +2634,18 @@ static void DLOpen(const FunctionCallbackInfo<Value>& args) {
     }
 
     // NOTE: `mp` is allocated inside of the shared library's memory, calling
-    // `uv_dlclose` will deallocate it
-    uv_dlclose(&lib);
+    // `dlclose` will deallocate it
+    dlib.Close();
     env->ThrowError(errmsg);
     return;
   }
   if (mp->nm_flags & NM_F_BUILTIN) {
-    uv_dlclose(&lib);
+    dlib.Close();
     env->ThrowError("Built-in module self-registered.");
     return;
   }
 
-  mp->nm_dso_handle = lib.handle;
+  mp->nm_dso_handle = static_cast<void*>(dlib);
```

Comment:
This looks confusing.  I'd do away with the cast operator overloading and replace it with a `dlib.handle()` call.

Since it's a small, one-off class, making it a simple struct and accesssing data members directly is acceptable too, IMO.  That will let you shrink it some more.